modules/tensor_mechanics/test/tests/cohesive_zone_model/ad_czm.i

```diff
@@ -0,0 +1,182 @@
+[Mesh]
+  [msh]
+    type = FileMeshGenerator
+    file = patch_mesh.e
+  []
+  [split]
+    type = BreakMeshByBlockGenerator
+    input = msh
+  []
+  [add_surfaces]
+    type = SideSetsFromNormalsGenerator
+    input = split
+    normals = '0  0  1
+               0  1  0
+               1  0  0
+               0  0 -1
+               0 -1  0
+              -1  0  0'
+    fixed_normal = true
+    new_boundary = 'z1 y1 x1 z0 y0 x0'
+  []
+[]
+
+[GlobalParams]
+  displacements = 'disp_x disp_y disp_z'
+  large_kinematics = true
+[]
+
+[Modules]
+  [TensorMechanics]
+    [Master]
+      [all]
+        strain = FINITE
+        add_variables = true
+        new_system = true
+      []
+    []
+    [CohesiveZoneMaster]
+      [interface]
+        boundary = 'interface'
+        strain = SMALL
+        use_automatic_differentiation = true
+      []
+    []
+  []
```

Comment:
Looks like you can mix the non-AD TM and AD CZM.

Reply:
yep